File: sample/src/main/java/com/readdle/swiftjava/sample/JNICore.java
package com.readdle.swiftjava.sample;

import com.readdle.codegen.anotation.SwiftGetter;
import com.readdle.codegen.anotation.SwiftReference;

@SwiftReference
public class JNICore {

    // Swift JNICore private native pointer
    private long nativePointer = 0L;

    // Swift JNICore private constructor
    // Should be private. Don't call this constructor from Java!
    private JNICore() {}

    // Swift JNICore release method
    private native void release();

    public native void dumpReferenceTables();

    @SwiftGetter("instance")
    public static native JNICore getJNI();

}

File: sample/src/main/java/com/readdle/swiftjava/sample/JNICore.kt
package com.readdle.swiftjava.sample

import com.readdle.codegen.anotation.SwiftGetter
import com.readdle.codegen.anotation.SwiftReference
import java.lang.annotation.Native

@SwiftReference
class JNICore private constructor() {

    // Swift JNICore private native pointer
    @Native
    private val nativePointer = 0L

    // Swift JNICore release method
    private external fun release()

    external fun dumpReferenceTables()

    companion object {

        @JvmStatic
        val JNI: JNICore?
            @SwiftGetter("instance")
            external get
    }

}
